runtime/ztests/expr/sql-substring.yaml

```diff
@@ -0,0 +1,15 @@
+spq: yield SUBSTRING(this FROM 4 FOR 2), SUBSTRING(this FROM 4), SUBSTRING(this FOR 3)
```

Comment:
Nit: Put this in compiler/ztests/sql/ since most (all?) of the SQL tests are in there.

Reply:
ok